lib/graphlient/adapters/http/adapter.rb

```diff
@@ -14,9 +14,25 @@ def headers
           options[:headers] if options
         end
 
+        def http_options
+          return {} unless options
+
+          options[:http_options] || {}
+        end
+
         def execute(*)
           raise NotImplementedError
         end
+
+        private
+
+        def configure_http_options(client_options)
+          http_options.each do |k, v|
+            set_method_name = "#{k}="
+
+            client_options.send(set_method_name, v) if client_options.respond_to?(set_method_name)
```

Comment:
This needs to be failing loudly, we want callers to know that they are trying to set an option that is not supported. Specialize an exception, rescue `NameError` here and raise that exception.

Reply:
Rescuing `NoMethodError` and re-raise a new `Graphlient::Errors::HttpOptionsError` now. Thx for all the feedback!